backend/tracim_backend/applications/upload_permissions/schema.py

```diff
@@ -0,0 +1,185 @@
+import cgi
+import typing
+
+import marshmallow
+from marshmallow import post_load
+from marshmallow.validate import OneOf
+
+from tracim_backend.app_models.validator import bool_as_int_validator
+from tracim_backend.app_models.validator import strictly_positive_int_validator
+from tracim_backend.applications.upload_permissions.models import UploadPermissionType
+from tracim_backend.applications.upload_permissions.validators import (
+    upload_permission_email_validator,
+)
+from tracim_backend.applications.upload_permissions.validators import (
+    upload_permission_password_validator,
+)
+from tracim_backend.lib.utils.utils import DATETIME_FORMAT
+from tracim_backend.views.core_api.schemas import StrippedString
+from tracim_backend.views.core_api.schemas import UserDigestSchema
+from tracim_backend.views.core_api.schemas import WorkspaceIdPathSchema
+
+
+class UploadFiles(object):
+    def __init__(self, **files: cgi.FieldStorage) -> None:
+        self.files = files.values()
+
+
+class UploadFilesSchema(marshmallow.Schema):
+    """
+    schema to handle up to 10 files
+    """
+
+    file1 = marshmallow.fields.Raw(required=False, description="a file")
+    file2 = marshmallow.fields.Raw(required=False, description="a file")
+    file3 = marshmallow.fields.Raw(required=False, description="a file")
+    file4 = marshmallow.fields.Raw(required=False, description="a file")
+    file5 = marshmallow.fields.Raw(required=False, description="a file")
+    file6 = marshmallow.fields.Raw(required=False, description="a file")
+    file7 = marshmallow.fields.Raw(required=False, description="a file")
+    file8 = marshmallow.fields.Raw(required=False, description="a file")
+    file9 = marshmallow.fields.Raw(required=False, description="a file")
+    file10 = marshmallow.fields.Raw(required=False, description="a file")
```

Comment:
10 files limitation is here by specification ? Or it is a "hack" to deal with hapic or something else ?

Reply:
Just a temporary hack to not have to make specific code to deal with unlimited number of file.

Reply:
Does exist an issue to solve that ? Frontend code will limit files number to 10 too ?

Reply:
@gbonaspetti @basilelegal @Skylsmoi Take care about this current limitation: 10 files maximum

Reply:
After discussion with @gbonaspetti @inkhey @PhilippeAccorsi backend will manage *n* files : by not using hapic yet on input files, no need to limit file number.

Reply:
Workaround as been done. fix for hapic related to issue #2209.